### PR TITLE
Frontend/Feature/ProfileCompleteReminder and FavouriteFcm

### PR DESCRIPTION
**Pull request by ROHIT-43**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added a pop to remind the driver for completing the profile and Driver will get fcm(Notification) whenever the user favourite's him.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

Manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/app/src/main/res/drawable/ny_ic_person_with_heart.png
Comment (Shikhar-24) [inline Frontend/android-native/app/src/main/res/drawable/ny_ic_person_with_heart.png:1]: compress bro

Comment (prakharritik) [inline Frontend/android-native/app/src/main/res/drawable/ny_ic_person_with_heart.png:1]: try to reduce the size

File: Frontend/ui-common/src/Components/PopUpModal/View.purs [hunk 1]
```diff
@@ -303,6 +302,11 @@ view push state =
             , onAnimationEnd (\_-> void $ pure $ startLottieProcess state.coverLottie.config )(const NoAction)
             ]
           ]
+
+        , case state.completeProfileLayout of
+            Just layout -> layout
+            Nothing -> textView [ visibility GONE]
```
Comment (prakharritik) [inline Frontend/ui-common/src/Components/PopUpModal/View.purs:308]: use noView from Mobility Prelude

File: Frontend/ui-common/src/Engineering/Helpers/Utils.js [hunk 2]
```diff
@@ -10,6 +10,11 @@ export const saveToLocalStoreImpl = function(key) {
   };
 }
 
+export const compareDate = function (date1, date2) {
+  console.log("comparing : ", date1, date2);
```
Comment (prakharritik) [inline Frontend/ui-common/src/Engineering/Helpers/Utils.js:14]: is this log required?

Comment (prakharritik) [reply Frontend/ui-common/src/Engineering/Helpers/Utils.js:14]: We have this function already

File: Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Handler.purs [hunk 3]
```diff
@@ -19,6 +20,8 @@ driverCompleteProfileScreen = do
   action <- lift $ lift $ runScreen $ DriverCompleteProfileScreen.screen state.driverCompleteProfileScreen (GlobalState state)
   case action of
     SubmitRes updatedState -> do
+      let completedModules = getValueBtwRange ((if updatedState.data.pledge == [] then 0 else 1) + (if updatedState.data.aspirations == [] then 0 else 1) + (if updatedState.data.drivingSince == Nothing then 0 else 1) + (if updatedState.data.homeTown == Nothing then 0 else 1) + (if updatedState.data.vehicalOffer == [] then 0 else 1) + (if updatedState.data.uploadedImagesIds == [] then 0 else 1)) 0 6 0 4
```
Comment (Shikhar-24) [inline Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Handler.purs:23]: same use **null** and **isNothing**

File: Frontend/ui-common/src/Screens/DriverProfileScreen/View.purs [hunk 4]
```diff
@@ -520,8 +526,10 @@ driverInfoCard push state =
             textView $
             [ height WRAP_CONTENT
             , width WRAP_CONTENT
-            , text $ "🌏    " <> getString I_SPEAK <> " " <> joinWith "" (DA.mapWithIndex(\index item -> do
-                if index == 0 then item else if index /= 0 && index /= (DA.length state.data.languages) - 1 then ", " <> item else getString AND <> item
+            , textFromHtml $ "🌏     " <> getString I_SPEAK <> " " <> joinWith "" (DA.mapWithIndex(\index item -> do
```
Comment (prakharritik) [inline Frontend/ui-common/src/Screens/DriverProfileScreen/View.purs:529]: move to let /where

File: Frontend/ui-customer/src/Helpers/Ride.purs [hunk 5]
```diff
@@ -346,223 +346,235 @@ checkRideStatus rideAssigned prioritizeRating = do
                 setValueToLocalStore TRACKING_ENABLED "True"
                 modifyScreenState $ HomeScreenStateType (\homeScreen → homeScreen{props{isSpecialZone = true,isInApp = true, isOtpRideFlow = true }}) else
                 pure unit
-      else if ((getValueToLocalStore RATING_SKIPPED) == "false" ) then do
+      else if (getValueToLocalStore RATING_SKIPPED /= "__failed" && getValueToLocalStore RATING_SKIPPED /="null" && getValueToLocalStore RATING_SKIPPED /= "true") then do
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Helpers/Ride.purs:349]: can be done with notElem

File: Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/View.purs [hunk 6]
```diff
@@ -103,7 +103,7 @@ view push state  =
     $ relativeLayout[
           width MATCH_PARENT
         , height MATCH_PARENT
-        , padding $ Padding 0 EHC.safeMarginTop 0 EHC.safeMarginBottom
+        , padding $ Padding 0 EHC.safeMarginTop 0 (if (state.isKeyBoardOpen) then 0 else EHC.safeMarginBottom)
```
Comment (Shikhar-24) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/View.purs:106]: store inside a variable `(if (state.isKeyBoardOpen) then 0 else EHC.safeMarginBottom)`

File: Frontend/ui-customer/src/Helpers/Ride.purs [hunk 7]
```diff
@@ -346,223 +346,235 @@ checkRideStatus rideAssigned prioritizeRating = do
...
+                  setValueToLocalStore TEST_POLLING_INTERVAL "8000.0"
+                  setValueToLocalStore TEST_POLLING_COUNT "22"
+                  setValueToLocalStore CONFIRM_QUOTES_POLLING_COUNT "100"
+                  pure unit
+                  else pure unit
+                when (isNothing currRideListItem.rideRating) $ do
+                  when (length listResp.list > 0) $ do
+                    let 
+                      fareProductType = getFareProductType ((resp.bookingDetails) ^. _fareProductType)
+                      (RideBookingAPIDetails bookingDetails) = resp.bookingDetails
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Helpers/Ride.purs:385]: these variables can be moved outside if else as they have been repeated

File: Frontend/ui-customer/src/Helpers/Ride.purs [hunk 8]
```diff
@@ -346,223 +346,235 @@ checkRideStatus rideAssigned prioritizeRating = do
...
+                                  }
+                                , rideRatingState
+                                    { driverName = currRideListItem.driverName
+                                    , rideId = currRideListItem.id
+                                    , distanceDifference = differenceOfDistance
+                                    , rideStartTime = case currRideListItem.rideStartTime of
+                                                        Just startTime -> (convertUTCtoISC startTime "h:mm A")
+                                                        Nothing        -> ""
+                                    , rideEndTime = case currRideListItem.rideEndTime of
+                                                      Just endTime   -> " " <>(convertUTCtoISC endTime "h:mm A")
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Helpers/Ride.purs:541]: use maybe function wherever such pattern is there

File: Frontend/ui-customer/src/Screens/CustomerUtils/FavouriteDriverTrips/View.purs [hunk 9]
```diff
@@ -67,6 +67,7 @@ getFavouriteDriverTrips push state = do
   (favouriteDriversResp) <- lift $ lift $ Remote.getFavouriteDriverTrips "5" "0" "false" state.data.driverNumber
   case favouriteDriversResp of
     Right resp -> do
+      let _ = spy "getFavouriteDriverTripsResp" resp
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Screens/CustomerUtils/FavouriteDriverTrips/View.purs:70]: remove spy

File: Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/ComponentConfig.purs [hunk 10]
```diff
@@ -25,7 +28,7 @@ primaryButtonConfig state = PrimaryButton.config
         }
     , alpha = if state.ratingCard.recordAudioState.isRecording == true then 0.5 else 1.0
     , background = Color.black900
-    , margin = (Margin 0 0 0 20)
+    , margin = (MarginBottom (if (state.isKeyBoardOpen) then 5 else 15))
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/ComponentConfig.purs:31]: remove redundant brackets

File: Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/ComponentConfig.purs [hunk 11]
```diff
@@ -40,15 +43,24 @@ primaryButtonConfigForCarousel state = PrimaryButton.config
...
     , rippleColor = Color.rippleShade
-    , isClickable = state.customerIssue.buttonActive
+    , isClickable = state.customerIssue.buttonActive || isOnlyDemandExtraTollIssue state
     }
 
+isOnlyDemandExtraTollIssue :: RiderRideCompletedScreenState -> Boolean
+isOnlyDemandExtraTollIssue state = do
+    let bannerConfigs = issueReportBannerConfigs state
+    if (DA.length bannerConfigs) == 1 then 
+        case bannerConfigs DA.!! 0 of
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/ComponentConfig.purs:59]: use maybe

File: Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/View.purs [hunk 12]
```diff
@@ -205,25 +205,15 @@ priceAndDistanceUpdateView state push =
...
                 ]
-            ]
-          , linearLayout[
-              width WRAP_CONTENT
-            , height WRAP_CONTENT
-            , visibility $ boolToVisibility $ state.showSafetyCenter
-            ][
-              sosButtonView state push  
-            ]
+            , linearLayout[weight 1.0][]
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/View.purs:215]: layoutWithWeight in mobility prelude

File: Frontend/ui-driver/src/Helpers/Utils.purs [hunk 13]
```diff
@@ -1107,6 +1108,15 @@ getVehicleVariantName variant =
...
                   UnKnown -> ""
 
+defaultNotificationBody :: NotificationBody
+defaultNotificationBody = {
+  title : "",
+  message : ""
+}
+
+reverse' :: String -> String
+reverse' = fromCharArray <<< DA.reverse <<< toCharArray
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Helpers/Utils.purs:1118]: Why is this required

File: Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Handler.purs (hunk 3 above)
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Handler.purs:23]: can be refactored

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 14]
```diff
@@ -497,6 +521,11 @@ eval (GoToButtonClickAC PrimaryButtonController.OnClick) state = do
     pure $ toggleBtnLoader "GotoClick" true
     updateAndExit state{ data { driverGotoState { savedLocationsArray = []}}} $ LoadGotoLocations state{ data { driverGotoState { savedLocationsArray = []}}}
 
+eval (ProfileDataAPIResponseAction res) state = do 
+  let DriverProfileDataRes resp = res 
+  continue state{data{completingProfileRes{
+    completed = getValueBtwRange ((if resp.pledges == [] then 0 else 1) + (if resp.aspirations == [] then 0 else 1) + (if resp.drivingSince == Nothing then 0 else 1) + (if resp.hometown == Nothing then 0 else 1) + (if resp.vehicleTags == [] then 0 else 1) + (if resp.otherImageIds == [] then 0 else 1)) 0 6 0 4}}}
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:527]: Refactor this

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 15]
```diff
@@ -715,16 +744,16 @@ eval (KeyboardCallback keyBoardState) state = do
...
     void $ pure $ setValueToLocalStore IS_DRIVER_AT_PICKUP "true"
     continueWithCmd newState [ pure if (not state.data.activeRide.notifiedCustomer) then NotifyAPI else AfterRender]
   else if (checkNotificationType notificationType ST.DRIVER_REACHED_DESTINATION && state.props.currentStage == ST.RideStarted && (not state.data.activeRide.notifiedReachedDestination)) then do
     continueWithCmd state [pure if (not state.data.activeRide.notifiedReachedDestination) then NotifyReachedDestination else AfterRender]
-  else if (Array.any ( _ == notificationType) [show ST.CANCELLED_PRODUCT, show ST.DRIVER_ASSIGNMENT, show ST.RIDE_REQUESTED, show ST.DRIVER_REACHED, show ST.TRIP_STARTED, show ST.EDIT_LOCATION]) then do
-      exit $ FcmNotification notificationType state{ props { specialZoneProps{ currentGeoHash = "" }} }
+  else if (Array.any ( _ == notificationType) [show ST.CANCELLED_PRODUCT, show ST.DRIVER_ASSIGNMENT, show ST.RIDE_REQUESTED, show ST.DRIVER_REACHED, show ST.TRIP_STARTED, show ST.EDIT_LOCATION, show ST.USER_FAVOURITE_DRIVER]) then do
+      exit $ FcmNotification notificationType (notificationBody) state{ props { specialZoneProps{ currentGeoHash = "" }} }
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:756]: redundant brackets

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 16]
```diff
@@ -2935,6 +2968,130 @@ checkRideCountAndTime tuple =
...
+              orientation VERTICAL
+          ][
+              linearLayout[
+                  height $ V 49,
+                  width $ V 98
+              ][
+                  linearLayout[
+                  height $ V 49,
+                  width $ V 49,
+                  background if per >= 50 then Color.blue900 else Color.white900,
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:3007]: these ifs can be refactored

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 17]
```diff
@@ -2935,6 +2968,130 @@ checkRideCountAndTime tuple =
...
+              padding $ Padding 12 2 12 2
+              ][
+              textView
+              $ [ text $ show ((state.data.completingProfileRes.completed*100)/4)<> "%"
+                  , width WRAP_CONTENT
+                  , height WRAP_CONTENT
+                  , color Color.white900
+                  , lineHeight "16"
+                  , textSize FontSize.a_15
+                  , fontStyle $ FontStyle.semiBold LanguageStyle
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:3089]: None of the fontstyle working for this?

**Review by prakharritik — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Engineering/Helpers/Utils.js (hunk 2 above)
Comment (ROHIT-43) [reply Frontend/ui-common/src/Engineering/Helpers/Utils.js:14]: yea we have that function but that was only available for user not driver i moved it to common.

File: Frontend/ui-customer/src/Helpers/Ride.purs (hunk 5 above)
Comment (ROHIT-43) [reply Frontend/ui-customer/src/Helpers/Ride.purs:349]: can you explain bit more about this ??

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs (hunk 17 above)
Comment (ROHIT-43) [reply Frontend/ui-driver/src/Screens/HomeScreen/View.purs:3089]: yes there was no fontstyle for this

File: Frontend/ui-common/src/Engineering/Helpers/Utils.js (hunk 2 above)
Comment (prakharritik) [reply Frontend/ui-common/src/Engineering/Helpers/Utils.js:14]: Can you remove this log

File: Frontend/android-native/app/src/main/res/drawable/ny_ic_person_with_heart.png
Comment (prakharritik) [inline Frontend/android-native/app/src/main/res/drawable/ny_ic_person_with_heart.png:1]: wont this be different for different variants?

File: Frontend/ui-customer/src/Helpers/Ride.purs [hunk 18]
```diff
@@ -346,223 +348,230 @@ checkRideStatus rideAssigned prioritizeRating = do
...
+                            }
                           }
-                      )
-                updateLocalStage RideCompleted
-              when (length listResp.list == 0 && not state.props.isSharedLocationFlow) $ do 
-                modifyScreenState $ HomeScreenStateType (\homeScreen → homeScreen{props{currentStage = HomeScreen}})
-                updateLocalStage HomeScreen
+                        )
+
+                    modifyScreenState
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Helpers/Ride.purs:475]: Can you move this to some different function, this one function is becoming large

File: Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/View.purs [hunk 19]
```diff
@@ -205,25 +207,15 @@ priceAndDistanceUpdateView state push =
...
           , gravity CENTER
           ][
-            linearLayout[ 
-              height WRAP_CONTENT
-            , width MATCH_PARENT
-            , weight 1.0
-            , gravity CENTER
-            , margin $ MarginLeft if state.showSafetyCenter then 60 else 0
-            ][ imageView
+              linearLayout[weight 1.0][]
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/RiderRideCompletedCard/View.purs:210]: use layoutWithWeight in mobility prelude

File: Frontend/ui-driver/src/Helpers/Utils.purs (hunk 13 above)
Comment (prakharritik) [reply Frontend/ui-driver/src/Helpers/Utils.purs:1119]: We can use from Data.array or data.string this is already present

File: Frontend/ui-driver/src/Flow.purs [hunk 20]
```diff
@@ -2444,8 +2444,10 @@ homeScreenFlow = do
   case action of             
     GO_TO_PROFILE_SCREEN updatedState -> do 
       liftFlowBT $ logEvent logField_ "ny_driver_profile_click"
-      modifyScreenState $ DriverProfileScreenStateType $ \driverProfileScreen -> driverProfileScreen { data { cachedVehicleCategory = fromMaybe ST.UnKnown $ RC.getCategoryFromVariant updatedState.data.vehicleType, cancellationRate = updatedState.data.cancellationRate}}
+      modifyScreenState $ DriverProfileScreenStateType $ \driverProfileScreen -> driverProfileScreen { data {profileCompletedModules = updatedState.data.completingProfileRes.completed, cachedVehicleCategory = fromMaybe ST.UnKnown $ RC.getCategoryFromVariant updatedState.data.vehicleType, cancellationRate = updatedState.data.cancellationRate}}
       driverProfileFlow
+    GO_TO_COMPLETE_PROFILE_SCREEN -> do
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Flow.purs:2449]: no need for do

File: Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Handler.purs [hunk 21]
```diff
@@ -3,22 +3,26 @@ module Screens.DriverCompleteProfileScreen.Handler where
...
 import Data.Maybe
+import Data.Array (null)
 
 driverCompleteProfileScreen :: FlowBT String DRIVER_COMPLETE_PROFILE_SCREEN_OUTPUT
 driverCompleteProfileScreen = do
   (GlobalState state) <- getState
   action <- lift $ lift $ runScreen $ DriverCompleteProfileScreen.screen state.driverCompleteProfileScreen (GlobalState state)
   case action of
     SubmitRes updatedState -> do
+      let completedModules = getValueBtwRange ((if (null updatedState.data.pledge) then 0 else 1) + (if (null updatedState.data.aspirations) then 0 else 1) + (if (isNothing updatedState.data.drivingSince) then 0 else 1) + (if (isNothing updatedState.data.homeTown) then 0 else 1) + (if (null updatedState.data.vehicalOffer) then 0 else 1) + (if (null updatedState.data.uploadedImagesIds) then 0 else 1)) 0 6 0 4
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Handler.purs:24]: boolToInt in mobilityPrelude

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 22]
```diff
@@ -499,6 +523,11 @@ eval (GoToButtonClickAC PrimaryButtonController.OnClick) state = do
     pure $ toggleBtnLoader "GotoClick" true
     updateAndExit state{ data { driverGotoState { savedLocationsArray = []}}} $ LoadGotoLocations state{ data { driverGotoState { savedLocationsArray = []}}}
 
+eval (ProfileDataAPIResponseAction res) state = do 
+  let DriverProfileDataRes resp = res 
+  continue state{data{completingProfileRes{
+    completed = getValueBtwRange ((if Array.null resp.pledges then 0 else 1) + (if Array.null resp.aspirations then 0 else 1) + (if isNothing resp.drivingSince then 0 else 1) + (if isNothing resp.hometown then 0 else 1) + (if Array.null resp.vehicleTags then 0 else 1) + (if Array.null resp.otherImageIds then 0 else 1)) 0 6 0 4}}}
```
Comment (prakharritik) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:529]: boolToInt from mobilityPrelude

**Review by prakharritik — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Components/PopUpModal/View.purs [hunk 23]
```diff
@@ -303,6 +302,11 @@ view push state =
             , onAnimationEnd (\_-> void $ pure $ startLottieProcess state.coverLottie.config )(const NoAction)
             ]
           ]
+
+        , case state.completeProfileLayout of
+            Just layout -> layout
+            Nothing -> noView
```
Comment (Shikhar-24) [inline Frontend/ui-common/src/Components/PopUpModal/View.purs:308]: maybe

**Review by prakharritik — DISMISSED** [review]

File: Frontend/ui-customer/src/Screens/CustomerUtils/FavouriteDriverTrips/View.purs [hunk 24]
```diff
@@ -150,7 +150,7 @@ view push state =
                     ] [ imageView
                         [ imageWithFallback $ fetchImage FF_ASSET "ic_vehicle_side" 
                         , width $ V 35
-                        , visibility VISIBLE
+                        , visibility GONE
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/CustomerUtils/FavouriteDriverTrips/View.purs:153]: Add a comment why it is removed

File: Frontend/ui-customer/src/Resources/Localizable/Types.purs [hunk 25]
```diff
@@ -1304,6 +1304,8 @@ data STR
   | WAS_TOLL_EXP_SMOOTH 
   | WAS_TOLL_EXP_SMOOTH_DESC  
   | WAS_DRIVER_HELPFUL 
+  | Demand_Extra_Toll_Amount
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Resources/Localizable/Types.purs:1307]: Use capital case

**Review by prakharritik — DISMISSED** [review]

File: Frontend/ui-common/src/Screens/DriverProfileScreen/View.purs [hunk 26]
```diff
@@ -495,6 +501,11 @@ driverStats push state =
 driverInfoCard :: forall w. (Action -> Effect Unit) -> DriverProfileScreenCommonState -> PrestoDOM (Effect Unit) w
 driverInfoCard push state =
     let drivingYears = (runFn2 JB.differenceBetweenTwoUTC (EHC.getCurrentUTC "") state.data.onboardedAt)
+        textForLanguages = "🌏     " <> getString I_SPEAK <> " " <> joinWith "" (DA.mapWithIndex(\index item -> do
+            if index == 0 then "<b>" <> item <> "</b>"
```
Comment (prakharritik) [inline Frontend/ui-common/src/Screens/DriverProfileScreen/View.purs:505]: you can make a function in let to create "<b>" <> item <> "</b>"

File: Frontend/ui-customer/src/Helpers/Ride.purs [hunk 27]
```diff
@@ -346,223 +348,136 @@ checkRideStatus rideAssigned prioritizeRating = do
...
+                                spLocationName = resp.specialLocationName,
+                                addressWard = bookingLocationAPIEntity.ward,
+                                providerType = maybe CTA.ONUS (\valueAdd -> if valueAdd then CTA.ONUS else CTA.OFFUS) resp.isValueAddNP,
+                                rentalData 
+                                    { finalDuration = (fromMaybe 0 resp.duration) / (60)
+                                    , finalDistance = (fromMaybe 0 ride.chargeableRideDistance)/1000
+                                    , baseDuration = (fromMaybe 0 resp.estimatedDuration) / (60*60)
+                                    , baseDistance = (fromMaybe 0 resp.estimatedDistance) / 1000
+                                    , rideStartedAt = case currRideListItem.rideStartTime of
+                                                        Just startTime -> (convertUTCtoISC startTime "h:mm A")
```
Comment (prakharritik) [inline Frontend/ui-customer/src/Helpers/Ride.purs:445]: use maybe

**Review by Vignesh-772 — APPROVED** [review]

**Review by Vignesh-772 — APPROVED** [review]
